crates/ibc/src/core/ics04_channel/error.rs

```diff
@@ -12,354 +12,242 @@ use crate::signer::SignerError;
-
-        InvalidChannelState
-            { channel_id: ChannelId, state: State }
-            | e | {
-                format_args!(
-                    "Channel {0} should not be state {1}",
-                    e.channel_id, e.state)
-            },
-
-        ChannelClosed
-            { channel_id: ChannelId }
-            | e | {
-                format_args!(
-                    "Channel {0} is Closed",
-                    e.channel_id)
-            },
-
-        ChanOpenAckProofVerification
-            | _ | { "Handshake proof verification fails at ChannelOpenAck" },
-
-        PacketCommitmentNotFound
-            { sequence: Sequence }
-            | e | {
-                format_args!(
-                    "Commitment for the packet {0} not found",
-                    e.sequence)
-            },
-
-        IncorrectPacketCommitment
-            { sequence: Sequence }
-            | e | {
-                format_args!(
-                    "The stored commitment of the packet {0} is incorrect",
-                    e.sequence)
-            },
-
-        PacketReceiptNotFound
-            { sequence: Sequence }
-            | e | {
-                format_args!(
-                    "Receipt for the packet {0} not found",
-                    e.sequence)
-            },
-
-        PacketAcknowledgementNotFound
-            { sequence: Sequence }
-            | e | {
-                format_args!(
-                    "Acknowledgment for the packet {0} not found",
-                    e.sequence)
-            },
-
-        MissingNextAckSeq
-            { port_id: PortId, channel_id: ChannelId }
-            | e | {
-                format_args!("Missing sequence number for ack packets on port {0} and channel {1}",
-                             e.port_id,
-                             e.channel_id)
-            },
-
-        ProcessedTimeNotFound
-            {
-                client_id: ClientId,
-                height: Height,
-            }
-            | e | {
-                format_args!(
-                    "Processed time for the client {0} at height {1} not found",
-                    e.client_id, e.height)
-            },
-
-        ProcessedHeightNotFound
-            {
-                client_id: ClientId,
-                height: Height,
-            }
-            | e | {
-                format_args!(
-                    "Processed height for the client {0} at height {1} not found",
-                    e.client_id, e.height)
-            },
-
-        RouteNotFound
-            | _ | { "route not found" },
-
-        ImplementationSpecific
-            | _ | { "implementation specific error" },
-
-        AppModule
-            { description: String }
-            | e | {
-                format_args!(
-                    "application module error: {0}",
-                    e.description)
-            },
+#[derive(Debug, Display)]
+pub enum ChannelError {
+    /// ICS03 connection error
```

Comment:
```suggestion
    /// connection error: {0}
```
As discussed in other comment. We should change the other variants analogously